ml-dsa/Cargo.toml

```diff
@@ -38,10 +38,13 @@ hex = { version = "0.4.3", features = ["serde"] }
 hex-literal = "0.4.1"
 pkcs8 = { version = "=0.11.0-rc.1", features = ["pem"] }
 proptest = "1"
-rand = "0.8.5"
+rand = "0.9"
 serde = { version = "1.0.215", features = ["derive"] }
 serde_json = "1.0.132"
 
 [[bench]]
 name = "ml_dsa"
 harness = false
+
+[profile.dev]
+opt-level = 2
```

Comment:
This is what caused the test failure: without it the stack seems to overflow.

Might suggest a case for `#[inline]` or `#[inline(always)]`